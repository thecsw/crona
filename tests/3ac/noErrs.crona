fn:int(a:int){
	if (a > 0) {
		write a;
	}
	write 1;
}
