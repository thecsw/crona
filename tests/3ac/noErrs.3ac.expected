[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN fn LOCALS]
a (formal arg of 8)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END fn LOCALS]
fun_fn:     enter fn
            getarg 1 [a]
            [tmp0] := 0
            [tmp1] := [a] GT64 [tmp0]
            IFZ [tmp1] GOTO lbl_1
            WRITE [a]
lbl_1:      nop
            WRITE 1
lbl_0:      leave fn
